prepare patch for plugins's Makefile.am
find . -name "Makefile.am" -exec sed -i "s@libdir = \$(pkgdatadir)/plugins@libdir = \@plugins_dir\@@" '{}' ';'

diff -Nur o.codeblocks-1.0rc2/acinclude.m4 n.codeblocks-1.0rc2/acinclude.m4
--- o.codeblocks-1.0rc2/acinclude.m4	2005-12-08 17:10:24.454151072 +0100
+++ n.codeblocks-1.0rc2/acinclude.m4	2005-12-08 17:10:13.204861224 +0100
@@ -228,3 +228,20 @@
 fi
 
 ])
+
+AC_DEFUN([CODEBLOCKS_PLUGINS_DIR],
+	[
+	if test "x$libdir" = "xNONE/lib" ; then
+		plugins_dir="/usr/local/lib/codeblocks/plugins"
+	else
+		plugins_dir="${libdir}/codeblocks/plugins"
+	fi
+	AC_ARG_WITH(plugins-dir,
+				AC_HELP_STRING([--with-plugins-dir=DIR],
+								[Path the plugins directory (default is "$plugins_dir")]),
+				,
+				with_plugins_dir="$plugins_dir")
+	AC_DEFINE_UNQUOTED(PLUGINDIR, "${with_plugins_dir}", [Code::Blocks plugins' directory])
+	AC_MSG_RESULT(PLUGINDIR: ${with_plugins_dir})
+	AC_SUBST(plugins_dir, ${with_plugins_dir})
+])
diff -Nur o.codeblocks-1.0rc2/codeblocks.pc.in n.codeblocks-1.0rc2/codeblocks.pc.in
--- o.codeblocks-1.0rc2/codeblocks.pc.in	2005-12-08 17:10:24.468148944 +0100
+++ n.codeblocks-1.0rc2/codeblocks.pc.in	2005-12-08 17:10:13.224858184 +0100
@@ -2,7 +2,7 @@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-plugindir=@datadir@/@PACKAGE@/plugins
+plugindir=@libdir@/@PACKAGE@/plugins
 sharedir=@datadir@
 
 Name: codeblocks
diff -Nur o.codeblocks-1.0rc2/configure.in n.codeblocks-1.0rc2/configure.in
--- o.codeblocks-1.0rc2/configure.in	2005-12-08 17:10:24.469148792 +0100
+++ n.codeblocks-1.0rc2/configure.in	2005-12-08 17:10:13.227857728 +0100
@@ -18,6 +18,7 @@
 AC_PROG_AWK
 
 CODEBLOCKS_SETUP_FOR_TARGET
+CODEBLOCKS_PLUGINS_DIR
 
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
diff -Nur o.codeblocks-1.0rc2/src/plugins/astyle/Makefile.am n.codeblocks-1.0rc2/src/plugins/astyle/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/astyle/Makefile.am	2005-12-08 17:10:24.740107600 +0100
+++ n.codeblocks-1.0rc2/src/plugins/astyle/Makefile.am	2005-12-08 17:10:13.478819576 +0100
@@ -5,7 +5,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libastyle.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/classwizard/Makefile.am n.codeblocks-1.0rc2/src/plugins/classwizard/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/classwizard/Makefile.am	2005-12-08 17:10:24.777101976 +0100
+++ n.codeblocks-1.0rc2/src/plugins/classwizard/Makefile.am	2005-12-08 17:10:13.491817600 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libclasswizard.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/codecompletion/Makefile.am n.codeblocks-1.0rc2/src/plugins/codecompletion/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/codecompletion/Makefile.am	2005-12-08 17:10:24.835093160 +0100
+++ n.codeblocks-1.0rc2/src/plugins/codecompletion/Makefile.am	2005-12-08 17:10:13.555807872 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libcodecompletion.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/compilergcc/Makefile.am n.codeblocks-1.0rc2/src/plugins/compilergcc/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/compilergcc/Makefile.am	2005-12-08 17:10:25.127048776 +0100
+++ n.codeblocks-1.0rc2/src/plugins/compilergcc/Makefile.am	2005-12-08 17:10:13.831765920 +0100
@@ -7,7 +7,7 @@
 
 AM_CXXFLAGS = -DDEPSLIB_UNIX
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libcompiler.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/contrib/codestat/Makefile.am n.codeblocks-1.0rc2/src/plugins/contrib/codestat/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/contrib/codestat/Makefile.am	2005-12-08 17:10:25.222034336 +0100
+++ n.codeblocks-1.0rc2/src/plugins/contrib/codestat/Makefile.am	2005-12-08 17:10:13.986742360 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libcodestat.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/contrib/profiler/Makefile.am n.codeblocks-1.0rc2/src/plugins/contrib/profiler/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/contrib/profiler/Makefile.am	2005-12-08 17:10:25.453999072 +0100
+++ n.codeblocks-1.0rc2/src/plugins/contrib/profiler/Makefile.am	2005-12-08 17:10:14.453671376 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libprofiler.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/debuggergdb/Makefile.am n.codeblocks-1.0rc2/src/plugins/debuggergdb/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/debuggergdb/Makefile.am	2005-12-08 17:10:26.501839776 +0100
+++ n.codeblocks-1.0rc2/src/plugins/debuggergdb/Makefile.am	2005-12-08 17:10:15.528507976 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libdebuggergdb.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/defaultmimehandler/Makefile.am n.codeblocks-1.0rc2/src/plugins/defaultmimehandler/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/defaultmimehandler/Makefile.am	2005-12-08 17:10:26.543833392 +0100
+++ n.codeblocks-1.0rc2/src/plugins/defaultmimehandler/Makefile.am	2005-12-08 17:10:15.568501896 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libdefaultmimehandler.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/pluginwizard/Makefile.am n.codeblocks-1.0rc2/src/plugins/pluginwizard/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/pluginwizard/Makefile.am	2005-12-08 17:10:26.577828224 +0100
+++ n.codeblocks-1.0rc2/src/plugins/pluginwizard/Makefile.am	2005-12-08 17:10:15.603496576 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libpluginwizard.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/todo/Makefile.am n.codeblocks-1.0rc2/src/plugins/todo/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/todo/Makefile.am	2005-12-08 17:10:26.621821536 +0100
+++ n.codeblocks-1.0rc2/src/plugins/todo/Makefile.am	2005-12-08 17:10:15.633492016 +0100
@@ -4,7 +4,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libtodo.la
 
diff -Nur o.codeblocks-1.0rc2/src/plugins/xpmanifest/Makefile.am n.codeblocks-1.0rc2/src/plugins/xpmanifest/Makefile.am
--- o.codeblocks-1.0rc2/src/plugins/xpmanifest/Makefile.am	2005-12-08 17:10:26.640818648 +0100
+++ n.codeblocks-1.0rc2/src/plugins/xpmanifest/Makefile.am	2005-12-08 17:10:15.675485632 +0100
@@ -2,7 +2,7 @@
 		-I$(top_srcdir)/src/sdk \
 		-I$(top_srcdir)/src/sdk/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+libdir = @plugins_dir@
 
 lib_LTLIBRARIES = libxpmanifest.la
 
diff -Nur o.codeblocks-1.0rc2/src/src/main.cpp n.codeblocks-1.0rc2/src/src/main.cpp
--- o.codeblocks-1.0rc2/src/src/main.cpp	2005-12-08 18:06:09.872255616 +0100
+++ n.codeblocks-1.0rc2/src/src/main.cpp	2005-12-08 18:06:13.733668592 +0100
@@ -23,6 +23,10 @@
 * $Date$
 */
 
+#ifdef HAVE_CONFIG_H
+    #include "config.h"
+#endif
+
 #include <sdk.h>
 #include "app.h"
 #include "main.h"
@@ -660,7 +664,9 @@
 
     PluginManager* m_PluginManager = Manager::Get()->GetPluginManager();
 
-    wxString path = CFG_READ(_T("data_path")) + _T("/plugins");
+    //XXX: FHS changes 
+    wxString path = PLUGINDIR;
+
     MSGMAN()->Log(_("Scanning for plugins in %s..."), path.c_str());
     int count = m_PluginManager->ScanForPlugins(path);
     MSGMAN()->AppendLog(_("Found %d plugins: "), count);
